lib/ts/recipe/emailverification/recipe.ts

```diff
@@ -164,4 +186,28 @@ export default class Recipe extends RecipeModule {
     isErrorFromThisRecipe = (err: any): err is STError => {
         return STError.isErrorFromSuperTokens(err) && err.fromRecipe === Recipe.RECIPE_ID;
     };
+
+    getEmailForUserId = async (userId: string, userContext: any): Promise<string> => {
+        if (this.config.getEmailForUserId !== undefined) {
+            const email = await this.config.getEmailForUserId(userId, userContext);
+            if (email !== undefined) {
+                return email;
+            }
+        }
+
+        for (const getEmailForUserId of this.getEmailForUserIdFuncsFromOtherRecipes) {
+            try {
+                const email = await getEmailForUserId(userId, userContext);
+                return email;
+            } catch {
+                // We ignore these, they should all be: Unknown User ID provided
+            }
+        }
+
+        throw new Error("Unknown User ID provided");
+    };
+
+    addGetEmailForUserIdFunc = (func: GetEmailForUserIdFunc): void => {
+        this.getEmailForUserIdFuncsFromOtherRecipes.push(func);
+    };
```

Comment:
Since this is an instance function, we should make `SessionRecipe.addClaimFromOtherRecipe` and others instance functions as well. Or else, make this a static function (keep things consistent).